common/src/main/scala/co/datamechanics/delight/common/Configs.scala

```diff
@@ -6,6 +6,22 @@ import scala.concurrent.duration._
 
 object Configs {
 
+  def isEdge(sparkConf: SparkConf): Boolean = {
+    sparkConf.getBoolean("spark.delight.edge", false)
+  }
+
+  def getDMAppId(sparkConf: SparkConf): String = {
+    val configPath = "spark.delight.dmAppId"
+    val dmAppId = sparkConf
+      .getOption(configPath)
+      .getOrElse {
+        val generatedDMAppId = generateDMAppId(sparkConf)
+        sparkConf.set(configPath, generatedDMAppId)
```

Comment:
This side-effect worries me a lot. Someone outside should take care of doing `set` or it should be a required config or `getDMAppId` should really have a different name.

Reply:
This will be used soon when we release our executor plugin
We'll need to share a same `dmAppId` across numerous process and the `sparkConf` is the easiest way to get everything sync together
The 1st element needing the `dmAppId` will generate it and set it in the config every other process will now pickit up fro there

Reply:
Mind to create an issue / task to track it?